inginious/frontend/plugins/upcoming_tasks/templates/coming_tasks.html

```diff
@@ -0,0 +1,134 @@
+                </a>
+            {% endfor %}
+        {% else %}
+            <a class="list-group-item list-group-item-action disabled submission-empty">{{ _("No submissions") }}</a>
+        {% endif %}
+    </div>
+    {{ template_helper.call('main_menu', template_helper=template_helper) | safe }}
+{% endblock %}
+
+
+
+{% block content %}
+
+<h2>{{ _("My Upcoming Tasks") }}
+<div class="pull-right">
+        <div class="btn-group btn-group-sm">
+            <button class="btn btn-warning" data-toggle="modal" data-target="#timeModal">
+                <i class="fa fa-exchange"></i> {{_("Switch time planner")}}
+            </button>
+        </div>
+    </div>
+</h2>
+
+<div class="modal fade" id="timeModal" tabindex="-1" role="dialog" aria-labelledby="timeModalLabel" aria-hidden="true">
+    <div class="modal-dialog" role="document">
+        <form method="post">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h5 class="modal-title" id="timeModalLabel">{{_("Modify time planner")}}</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body">
+                    <p>{{ _("This allows to display only the tasks whose deadline is in a certain temporal proximity.") }}</p>
+                    <label>{{ _("Number of days:") }}</label>
+                    <select class="form-control" id="new_time_planner" name="time_planner">
+                        {% for timeid in time_planner %}
+                            <option value="{{timeid}}">{{ timeid}}</option>
+                        {% endfor %}
+                    </select>
+                </div>
+                <div class="modal-footer">
+                    <button type="button" class="btn btn-secondary" data-dismiss="modal">{{_("Cancel")}}</button>
+                    <button type="submit" class="btn btn-warning">{{_("Save changes")}}</button>
+                </div>
+            </div>
+        </form>
+    </div>
+</div>
+
+
+<div class="list-group">
+	{%if open_courses %}
+        {% set username = user_manager.session_username() %}
+        {% for courseid, course in open_courses.items() %}
+                <div class="row">
+                    <div class="col-xs-12 col-md-8">
+					{{ course.get_name(user_manager.session_language()) }}
+					{{ my_render.render_upcoming_list(template_helper, course, tasks_data, tasks_list, my_render) | safe }}
+
+
+                    </div>
+                    <div class="col-xs-12 col-md-4">
+                        <i class="pull-right">
+                            {% if course.is_lti() %}
+                                {{ _("LTI course") }} -
```

Comment:
If LTI deadlines are displayed, a link to the external platform should be given instead. However, deadlines can and should rather be set at the external platform side. There may also be confusion as the course structure may be different between the two platforms. It can stay here, but I think it has a limited interest.

Reply:
I removed this part together with the fact showing ("Hidden course" on the right side) since anyway courses are filtered before being sent to the html and this scenario will never show up.
This pull-right col was simply copy pased from another html file but had actually no purpose here.